File: android/app/src/main/java/com/example/android/domain/model/Transaction.kt
package com.example.android.domain.model

data class Transaction(
    val id: String,
    val userId: String,
    val type: String, // "DEPOSIT" | "BUY_PACKAGE"
    val amount: Double,
    val packageId: String? = null,
    val status: String = "PENDING", // "SUCCESS" | "FAILED"
    val createdAt: Long = System.currentTimeMillis()
)

File: backend/src/domain/models/Transaction.js
const mongoose = require('mongoose');

const transactionSchema = new mongoose.Schema({
  _id: { type: String, required: true },
  user_id: { type: String, ref: 'User', required: true },
  type: { type: String, enum: ['DEPOSIT', 'BUY_PACKAGE'], required: true },
  amount: { type: Number, required: true },
  
  package_id: { type: String, ref: 'ProPackage', default: null },
  status: { type: String, enum: ['PENDING', 'SUCCESS', 'FAILED'], default: 'PENDING' }
}, { 
  timestamps: { createdAt: 'created_at', updatedAt: false } 
});

transactionSchema.index({ user_id: 1, created_at: -1 });

module.exports = mongoose.model('Transaction', transactionSchema);